### PR TITLE
Feature/binary

### PR DESCRIPTION
**Pull request by ivarconr**
This PR solved #150 

Because unleash-api now automatically checks and migrates the database on startup we don't have to confuse the user with setting up their own migrations.  

Using the binary is now as simple as: 

```bash
$ npm install unleash-server -g
$ unleash -d postgres://unleash_user:passord@localhost:5432/unleash -p 4242

Unleash started on port:4242
```

2016-09-29-224241_1011x171_scrot

### CODE REVIEW

File: packages/unleash-server/unleash.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+'use strict';
+
+process.env.NODE_ENV = 'production';
+
+const program = require('commander');
+const unleash = require('./server.js');
+
+
+program
+    .option('-p, --port <port>', 'The full port you want to start unleash on.')
```
Comment (sveisvei) [inline packages/unleash-server/unleash.js]: remove the "full" part

File: packages/unleash-server/package.json [hunk 2]
```diff
@@ -24,17 +24,19 @@
     "node": "6"
   },
   "main": "./server.js",
+  "bin": {
+    "unleash": "./unleash.js"
```
Comment (sveisvei) [inline packages/unleash-server/package.json]: could be nice to move the unleash.js to a bin folder?

File: packages/unleash-server/unleash.js (hunk 1 above)
Comment (ivarconr) [inline packages/unleash-server/unleash.js]: full as in drunk :dancer:

File: packages/unleash-api/migrator.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const DBMigrate = require('db-migrate');
+const path = require('path');
+
+function findUnleashApiRoot () {
+    try {
+        return path.dirname(require.resolve('unleash-api/package.json'));
+    } catch (e) {}
+    try {
+        return path.dirname(require.resolve('../unleash-api'));
```
Comment (sveisvei) [inline packages/unleash-api/migrator.js]: return path.dirname(require.resolve('../unleash-api/package.json'));

**Comment by sveisvei** [comment]
wfm! 👍

**Comment by ivarconr** [comment]
fixed. Whats really nice is that it should be fairly easy to add new hooks later, e.g. an option to disable db-migration, allow user to only do migrations and so on..

**Comment by sveisvei** [comment]
Merge it like its hot 💯

**Pull request merged**